### PR TITLE
[hal] Document `Queue::submit` ordering guarantees a bit.

### PR DESCRIPTION
**Pull request by jimblandy**

### CODE REVIEW

**Comment by jimblandy** [comment]
Hmm... I think this could be phrased better.

File: wgpu-hal/src/lib.rs [hunk 1]
```diff
@@ -553,6 +553,15 @@ pub trait Queue: WasmNotSendSync {
 
     /// Submits the command buffers for execution on GPU.
     ///
+    /// Command buffers execute in the order in which they appear in
+    /// `command_buffers`. All results produced by one buffer are visible to the
+    /// next.
+    ///
+    /// If the caller has ensured that one call to `Queue::submit` on a
```
Comment (JMS55) [inline wgpu-hal/src/lib.rs]: /// If `Queue::submit` is called on one `Queue`, and then again on a different queue,
    /// then all the command buffers from the first call will finish executing before any of the 
    /// command buffers from the second call begin.

Comment (jimblandy) [reply wgpu-hal/src/lib.rs]: Actually, there's no ordering between queues. I've rephrased to make this clearer.

**Comment by jimblandy** [comment]
This will conflict with the other hal doc PRs, but it's fine to review.

**Review by Wumpf — APPROVED** [review]
tough one. I had to spend quite a bit of time convince myself of this bit `all complete execution before any of the second submission's commands begin`, but from the looks of it we really do enforce that strictly.

**Comment by jimblandy** [comment]
> tough one. I had to spend quite a bit of time convince myself of this bit `all complete execution before any of the second submission's commands begin`, but from the looks of it we really do enforce that strictly.

Thank you for the close reading! I was unsure too, but I think we couldn't function without this requirement, because waiting for fence values only works if things actually complete in increasing fence value order. But see also #5559.

**Pull request merged**